pkg/cache/remote.go

```diff
@@ -22,11 +22,22 @@ type RemoteCache struct {
 // RemoteURL to hold remote host
 type RemoteURL string
 
+// Insecure for RPC remote host
+type Insecure bool
+
 // NewRemoteCache is the factory method for RemoteCache
-func NewRemoteCache(url RemoteURL, customHeaders http.Header) cache.ArtifactCache {
+func NewRemoteCache(url RemoteURL, insecure Insecure, customHeaders http.Header) (cache.ArtifactCache, error) {
```

Comment:
Why do we need to return an error?

Reply:
Good catch. Thanks.

That's a left over from some other implementation I tried.